dipy/data/tests/test_fetcher.py

```diff
@@ -67,6 +101,11 @@ def test_fetch_data():
 
         # Test that an error is raised when the md5 checksum of the download
         # file does not match the expected value
-        files = {"testfile.txt": (testfile_url, bad_md5)}
+        files = {"testfile.txt": (test_server_url, bad_md5)}
         npt.assert_raises(fetcher.FetcherError,
                           fetcher.fetch_data, files, tmpdir)
+
+        # stop local HTTP Server
+        server.shutdown()
+        # change to original working directory
```

Comment:
no actually force closing the thread is not necessary. server.shutdown() ends the server and the thread target function returns safely and the thread dies peacefully (based on what I understood about threads)